valohai_yaml/objs/task.py

```diff
@@ -0,0 +1,94 @@
+from enum import Enum
+from typing import Any, List, Optional, Union
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.objs.utils import (
+    check_type_and_listify,
+    consume_array_of,
+)
+
+
+class TaskParameterType(Enum):
+    """Represents a task parameter style definition."""
+
+    BAYESIAN_TPE = 'bayesian_tpe'
+    GREED_SEARCH = 'greed_search'
```

Comment:
is this supposed to be Grid Search? I don't think we have Greed search or have I missed something?

Reply:
Why did the greedy internet user become a search engine expert?
Because they wanted to hoard all the "greed-able" information!